### ISSUE TITLE
Package load failed for 'S4Vectors'

### ISSUE THREAD

**Issue by Zifeng-L** [Bioconductor/S4Vectors#65]
Recently, in order to install the dev version of singleR, I changed R 3.6.2 to R 4.0.0 and installed the Rtools40. When I load the package called S4Vectors, there's something wrong.

`
> library(S4Vectors)
Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter,
    Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int,
    pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply, union, unique,
    unsplit, which, which.max, which.min

Error: package or namespace load failed for ‘S4Vectors’:
 .onLoad failed in loadNamespace() for 'S4Vectors', details:
  call: validObject(.Object)
  error: invalid class “LLint” object: superclass "integer_OR_LLint" not defined in the environment of the object's class`

How can I deal with this problem? Thanks

**Comment by Zifeng1995** [comment]
In R3.6.2 the package can be loaded successfully. I copy the package to the dev R lib.  The package still cannot be loaded.

**Comment by PeteHaitch** [comment]
On R 4.0, it doesn't sound like you followed the [installation instructions](https://bioconductor.org/packages/devel/bioc/html/S4Vectors.html), i.e.

```R
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

# The following initializes usage of Bioc devel
BiocManager::install(version='devel')

BiocManager::install("S4Vectors")
```

**Comment by Zifeng1995** [comment]
> ```r
> BiocManager::install(version='devel')
> 
> BiocManager::install("S4Vectors")
> ```

Thanks for guiding me. I removed the package and followed the installation instructions, but the error still existed.
` 
> BiocManager::install(version='devel')
Bioconductor version 3.11 (BiocManager 1.30.10), R 4.0.0 Under development (Rtools 4.0) (2020-02-21 r77847)
> 
> BiocManager::install("S4Vectors")
Bioconductor version 3.11 (BiocManager 1.30.10), R 4.0.0 Under development (Rtools 4.0) (2020-02-21 r77847)
Installing package(s) 'S4Vectors'
trying URL 'https://bioconductor.org/packages/3.11/bioc/bin/windows/contrib/4.0gcc8/S4Vectors_0.25.12.zip'
Content type 'application/zip' length 2175345 bytes (2.1 MB)
downloaded 2.1 MB

package ‘S4Vectors’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\lizif\AppData\Local\Temp\RtmpIDjFZB\downloaded_packages
> library(S4Vectors)
Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter,
    Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int,
    pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply, union, unique,
    unsplit, which, which.max, which.min

Error: package or namespace load failed for ‘S4Vectors’:
 .onLoad failed in loadNamespace() for 'S4Vectors', details:
  call: validObject(.Object)
  error: invalid class “LLint” object: superclass "integer_OR_LLint" not defined in the environment of the object's class`

**Comment by mtmorgan** [comment]
LIkely other packages have not been installed correctly; does `BiocManager::valid()` indicate that some packages should be updated?

**Comment by Zifeng1995** [comment]
> LIkely other packages have not been installed correctly; does `BiocManager::valid()` indicate that some packages should be updated?

sorry for late. I just ran `BiocManager::valid()`, the result is as follows
`
> BiocManager::valid() 
[1] TRUE
Warning messages:
1: In readChar(...) : can only read in bytes in a non-UTF-8 MBCS locale
2: In readChar(...) : can only read in bytes in a non-UTF-8 MBCS locale
3: In readChar(...) : can only read in bytes in a non-UTF-8 MBCS locale
4: In readChar(...) : can only read in bytes in a non-UTF-8 MBCS locale
5: In readChar(...) : can only read in bytes in a non-UTF-8 MBCS locale` 

It seems that no packages need be updated.

**Comment by lawremi** [comment]
@mtmorgan Has the S4Vectors Windows binary been rebuilt on the repository?

**Comment by FelixErnst** [comment]
fyi: The original error message in the first post is very similar to what others (See slack, 3-4 weeks ago) and I have seen on the travis build system. Deleting the cache solved the issue there, but I don't know what will fix it on a local install.

**Comment by mtmorgan** [comment]
@hpages would let us know about binary rebuilds... just a version bump would do it...

**Comment by Zifeng1995** [comment]
Thank you all. So I could only wait？or do something else？

**Comment by hpages** [comment]
First we need to update R to the latest R 4.0 on our Windows builder tokay2. We had this scheduled for today but had to postpone to tomorrow for internal reasons. Then we'll bump S4Vectors version (and probably other core packages) in order to trigger propagation of new binaries. 

So the whole process will take at least a couple of days. In the mean time I strongly suggest that you install all your packages from source. Since you have Rtools, that should not be a problem for you. 

However, please note that Rtools 4.0 is still in beta and should be used only with the special R-testing build of R 4.0 at the moment, not if you're using the official Windows build of R 4.0 provided by CRAN. See: https://cran.r-project.org/bin/windows/Rtools/

**Comment by Zifeng1995** [comment]
I appreciate your help very much！！ps： I did use the Rtools4.0 with the R-testing build of R4.0，maybe I wasn't being clear. I should be very grateful if you would let me know the informations when it works.

**Comment by hpages** [comment]
BTW any reason you're using Rtools4.0 + R-testing?

Just to clarify we're still using Rtools35 and the official CRAN binary of R 4.0 on our Windows builder tokay2. We're not planning to switch to Rtools 4.0 + R-testing yet. So if you're using Rtools4.0 + R-testing, there is no guarantee that the binaries we produce will work for you, even after we update R on tokay2 and generate new binaries for S4Vectors (and other core packages). Maybe they'll work for you, maybe not.

In other words, if you want to be on the safe side, I would recommend that you stick to Rtools35 and the official CRAN binary of R 4.0. If for some reason you really want to use Rtools 4.0 + R-testing, you should install Bioconductor packages from source, at least for now.

**Comment by lawremi** [comment]
You need to install the package from source, i.e., pass `type="source"` to `install.packages()`.

**Comment by hpages** [comment]
An update on this: Lori (@lshep) updated tokay2 (our Windows build machine) with the latest R 4.0 this morning and I bumped the version of S4Vectors (to 0.25.11), IRanges (to 2.21.4) and XVector (to 0.27.1). Didn't do it for GenomicRanges since the package is currently broken so wouldn't propagate anyway.

Keep an eye on the [build report for Bioc devel](https://bioconductor.org/checkResults/3.11/bioc-LATEST/) Should get updated tomorrow (Friday) around 2:20-3 pm EST. Then the new binaries should become available via `BiocManager::install()` about 2-3h later. That's if everything goes as expected but who knows. The 1st build run after a fresh R re-install tends to be a little like playing lottery, especially on Windows and when updating to the latest R devel. Don't hold your breath ;-)

**Comment by Zifeng1995** [comment]
thanks a lot!!!!

**Comment by hpages** [comment]
No problem. But again, if you're using Rtools4.0 + R-testing, those new binaries are not for you. AFAIK nobody is producing Bioconductor binaries for Rtools4.0 + R-testing at the moment so you should really install from source. Did you try doing that?

**Comment by hpages** [comment]
New build report is here: https://bioconductor.org/checkResults/3.11/bioc-LATEST/ It is not pretty!

But wait! What? Oh great, looks like instead of bumping the version of S4Vectors yesterday, I went down (from 0.25.12 to 0.25.11).

I just tried again to bump the version (to 0.25.13).

**Comment by hpages** [comment]
S4Vectors 0.25.13 has finally propagated: https://bioconductor.org/packages/3.11/S4Vectors.

Can you install it and do `library(S4Vectors)`?
